components/num-util/src/fixed_decimal.rs

```diff
@@ -373,6 +377,135 @@ impl fmt::Display for FixedDecimal {
     }
 }
 
+impl FromStr for FixedDecimal {
+    type Err = Error;
+    fn from_str(input_str: &str) -> Result<Self, Self::Err> {
+        // input_str: the input string
+        // no_sign_str: the input string when the sign is removed from it
+        let no_sign_str: &str;
+        let mut is_negative = false;
+        if &input_str[0..1] == "-" {
+            is_negative = true;
+            no_sign_str = &input_str[1..];
+        } else {
+            no_sign_str = input_str;
+        }
+        // Compute length of each string once and store it, so if you use that multiple times,
+        //  you don't compute it multiple times
+        // has_dot: shows if your input has dot in it
+        // dot_index: gives the index of dot (after removing the sign)
+        let no_sign_str_len = no_sign_str.chars().count();
+        let mut has_dot = false;
+        let mut dot_index = no_sign_str_len;
+        // The following loop computes has_dot, dot_index, and also checks to see if all
+        // characters are digits and if you have at most one dot
+        // Note: Input of format 111_123 is detected as syntax error here
+        // Note: Input starting or ending with a dot is detected as syntax error here (Ex: .123, 123.)
+        for (i, c) in no_sign_str.chars().enumerate() {
```

Comment:
Optional: since this is expected to be an ASCII string, consider enumerating bytes instead of chars.

```rust
no_sign_str.bytes().enumerate()
```

Below, you might need to change things like `'.'` to `b'.'`.